includes/admin/functions-menu.php

```diff
@@ -50,18 +50,33 @@ function wpas_tickets_count() {
 
 	global $menu, $current_user;
 
-	if ( current_user_can( 'administrator' )
-		 && false === boolval( wpas_get_option( 'admin_see_all' ) )
-		 || ! current_user_can( 'administrator' )
+	$is_agent = current_user_can( 'administrator' )
+		&& false === boolval( wpas_get_option( 'admin_see_all' ) )
+		|| ! current_user_can( 'administrator' )
 			&& current_user_can( 'edit_ticket' )
-			&& false === boolval( wpas_get_option( 'agent_see_all' ) )
-	) {
+			&& false === boolval( wpas_get_option( 'agent_see_all' ) );
 
-		$agent = new WPAS_Member_Agent( $current_user->ID );
-		$count = $agent->open_tickets();
+	$count_cache = get_site_transient( 'wpas_tickets_counts' );
 
-	} else {
-		$count = count( wpas_get_tickets( 'open' ) );
+	if( !is_array($count_cache) )
+		$count_cache = array();
+
+	$agent_id = $is_agent ? $current_user->ID : 0;
+
+	if( !isset( $count_cache[$agent_id] ) )  {
```

Comment:
Why not starting with that check at the very beginning? Setting `$is_agent` and `$agent_id` doesn't bring any value up to this point (it's not being used). You could check the cache as a very first step, and then if there is no result go on with the rest of the processing.

```
$count_cache = get_site_transient( 'wpas_tickets_counts' );

if ( isset( $count_cache[ $agent_id ] ) ) {
    // No need to calculate
} else {
  // Process the rest
}
```

Reply:
`$is_agent` and `$agent_id` is actually used, check my code line 68 & 70. As in my previous